actors/datacap/src/lib.rs

```diff
@@ -451,15 +452,22 @@ where
     ) -> fvm_actor_utils::messaging::Result<Receipt> {
         // The Runtime discards some of the information from the syscall :-(
         let fake_gas_used = 0;
-        let fake_syscall_error_number = ErrorNumber::NotFound;
-        self.rt
-            .send(to, method, params.clone(), value.clone())
-            .map(|bytes| Receipt {
-                exit_code: ExitCode::OK,
-                return_data: bytes,
-                gas_used: fake_gas_used,
-            })
-            .map_err(|_| MessagingError::Syscall(fake_syscall_error_number))
+        let res = self.rt.send(to, method, params.clone(), value.clone());
+
+        let rec = match res {
+            Ok(bytes) => {
+                Receipt { exit_code: ExitCode::OK, return_data: bytes, gas_used: fake_gas_used }
+            }
+            Err(ae) => {
+                info!("datacap messenger failed: {}", ae.msg());
+                Receipt {
+                    exit_code: ae.exit_code(),
+                    return_data: RawBytes::default(),
+                    gas_used: fake_gas_used,
+                }
+            }
+        };
+        Ok(rec)
```

Comment:
Commentary: This never returns an Err now. But I think this makes sense: we _could_ return an error if the Runtime `send` method passed through the underlying syscall error instead of hiding it behind a client-side chosen exit code.